File: capy/src/e2e-cfun/java/dev/capylang/test/collection/TupleCollectionTest.java
package dev.capylang.test.collection;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class TupleCollectionTest {
    @Test
    void tupleWithFunctionsApplyFirst() {
        assertThat(TupleCollection.tupleWithFunctionsApplyFirst(7)).isEqualTo(8);
    }

    @Test
    void tupleWithFunctionsApplySecond() {
        assertThat(TupleCollection.tupleWithFunctionsApplySecond(7)).isEqualTo(14);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/collection/TupleCollection.test.js
const test = require('node:test');
const { assert, generatedModule } = require('../_support');

test('TupleCollection', () => {
    const tuple = generatedModule('dev/capylang/test/collection/TupleCollection.js');
    assert.equal(tuple.tupleWithFunctionsApplyFirst(7), 8);
    assert.equal(tuple.tupleWithFunctionsApplySecond(7), 14);
});
